packages/babel-parser/src/parser/expression.js

```diff
@@ -1120,6 +1120,17 @@ export default class ExpressionParser extends LValParser {
           );
         }
 
+        if (
+          this.match(tt.comma) &&
+          (this.lookahead().type === tt.name ||
+            this.lookahead().type === tt.ellipsis)
```

Comment:
We don't need the lookahead, since also `(a, ...b,) => {}` is invalid.

Reply:
@nicolo-ribaudo right, so basically we can do just:
```
if (this.match(tt.comma)) {
          this.raise(
            this.state.start,
            "A trailing comma is not permitted after the rest element",
          );
        }
```

Reply:
@nicolo-ribaudo so you want to leave only this type of error? 
`"A trailing comma is not permitted after the rest element"`

Reply:
We can combine the two conditionals and do the lookahead once?

Reply:
I agree with @existentialism. Btw, in the lookahead we should check for `}` and `]`.

Reply:
@existentialism @nicolo-ribaudo so maybe it's even better to add switch there

Reply:
Well, there are only two cases:
1. Trailing comma
2. Another parameter

Reply:
@nicolo-ribaudo what about `[` and `{`?

Reply:
@nicolo-ribaudo @existentialism So something like this

```
if (this.match(tt.comma)) {
    const nextTokenType = this.lookahead().type;
    const errorMessage =
    nextTokenType === tt.parenR
        ? "A trailing comma is not permitted after the rest element"
        : "Rest parameter must be last formal parameter";
    this.raise(this.state.start, errorMessage);
}
```

Reply:
👍

Reply:
Yes :+1: